Fix #18248 - Change password modal doesn't open

Odachannnn opened the pull request:
## Description
 
This pull request fixes change password modal doesn't open.

Fixes  #18248

Odachannnn committed:
Fix Change password modal doesn't open

Signed-off-by: Asako Oda <12odadao14@gmail.com>

williamdes commented on templates/header.twig:
```diff
@@ -58,5 +58,4 @@
     {{ include('modals/preview_sql_modal.twig') }}
     {{ include('modals/enum_set_editor.twig') }}
     {{ include('modals/create_view.twig') }}
-    {{ include('modals/change_password.twig') }}
```
You should probably keep this line, because in some cases the modal will not be part of the content

I am unsure about my comment, but it's worth checking if there is other ways to trigger this modal to open or not

Odachannnn replied:
@williamdes 
> I am unsure about my comment, but it's worth checking if there is other ways to trigger this modal to open or not

I checked it, and I found that I must not remove this line.
When I change the theme, the modal doesn't not open.


https://user-images.githubusercontent.com/63225126/226154197-67e35df1-834f-4ac6-8bb6-a79620b54e4e.mov

williamdes replied:
@MauricioFauth 
Because you pushed 01cf59bf8d1281264e32f3639ce8d8fd2dfeac90

MauricioFauth replied:
I did not see this discussion when I was reviewing this PR, otherwise I would have accepted it.

MauricioFauth replied:
The change I did doesn't have this issue.

williamdes replied:
okay, thanks for checking then

Odachannnn committed:
Fix Change password modal doesn't open

Revert including change password modal in header.twig and fix the condition to include it in index.twig

Signed-off-by: Asako Oda <12odadao14@gmail.com>

Odachannnn commented:
I'm sorry if force push is  not good after PR.
I did out of habit.

williamdes commented:
> I'm sorry if force push is not good after PR. I did out of habit.

No no, it's totally okay I prefer that than users that create me more work by closing an opening a new one because for some reason they can not force push

williamdes commented on templates/home/index.twig:
```diff
@@ -6,6 +6,10 @@
 
 {{ partial_logout|raw }}
 
+{%- if has_change_password_modal == false %}
```
```suggestion
{%- if not has_change_password_modal %}
```

williamdes commented on libraries/classes/Controllers/HomeController.php:
```diff
@@ -213,6 +213,9 @@ public function __invoke(ServerRequest $request): void
 
         $git = new Git($this->config->get('ShowGitRevision') ?? true);
 
+        $hasChangePasswordLink = ($GLOBALS['cfg']['Server']['auth_type'] ?? '') !== 'config'
```
Maybe suggest to #18274 to pick this change in their batch so it also does not add the modal globally

Odachannnn replied:
I'm not good at English, and I'm not used to contributing and contributing flow,
so I'd like to get a better understanding of your comment.

You mean that I should suggest to #18274 to change codes like this line, and write a comment?
I checked comments of the PR, you have already commented so.

williamdes reviewed (approved):
💯

Odachannnn committed:
replace an equal comparison operator with not  in home/index.twig

Co-authored-by: William Desportes <williamdes@wdes.fr>
